ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake

```diff
@@ -24,5 +24,14 @@ file(GLOB_RECURSE _source_files FOLLOW_SYMLINKS
 )
 if(_source_files)
   message(STATUS "Added test 'cppcheck' to perform static code analysis on C / C++ code")
-  ament_cppcheck()
+  # Get include paths for sources
+  set(_include_dirs "")
+  foreach(_src ${_source_files})
+    get_filename_component(_src_dir ${_src} DIRECTORY)
+    # Get parent directory since includes typically have the pattern 'package/source.hpp'
+    get_filename_component(_parent_dir ${_src_dir} DIRECTORY)
```

Comment:
This heuristic looks rather fragile to me. I can imagine cases where we have multiple levels of nesting where this might even lead to false results.

Reply:
I discovered an alternative command to this heuristic. Hopefully less fragile. 
aa40c90

Reply:
Setting global include directories is discouraged in favor of target specific properties. So this approach is unlikely going to work on the future.

Reply:
I'm not sure how else to infer the include directories here, (until cmake 3.7 when we can use [BUILDSYSTEM_TARGETS](https://cmake.org/cmake/help/v3.13/prop_dir/BUILDSYSTEM_TARGETS.html) to get `INCLUDE_DIRECTORIES` for each target).

Alternatively, affected packages can call the ament cppcheck macro explicitly, passing the appropriate include paths.

Reply:
> until cmake 3.7 when we can use `BUILDSYSTEM_TARGETS` to get `INCLUDE_DIRECTORIES` for each target

That sounds like the best path forward as soon as we decide to drop Xenial support.